internal/health/health.go

```diff
@@ -31,3 +45,153 @@ func healthResponse() func(w http.ResponseWriter, req *http.Request) {
 		fmt.Fprint(w, "OK")
 	}
 }
+
+func runFullControllersCycle(client client.Client, lg *logrus.Entry) func(w http.ResponseWriter, req *http.Request) {
+	return func(w http.ResponseWriter, req *http.Request) {
+		if err := runAddonsConfigurationControllerCycle(client, lg); err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+		if err := runClusterAddonsConfigurationControllerCycle(client, lg); err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+
+		w.WriteHeader(http.StatusOK)
+		fmt.Fprint(w, "OK")
+	}
+}
+
+func runAddonsConfigurationControllerCycle(client client.Client, lg *logrus.Entry) error {
+	probeName := "liveness-probe"
+	probeNamespace := "default"
+
+	addonsConfiguration := &v1alpha1.AddonsConfiguration{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      probeName,
+			Namespace: probeNamespace,
+		},
+		Spec: v1alpha1.AddonsConfigurationSpec{
+			CommonAddonsConfigurationSpec: v1alpha1.CommonAddonsConfigurationSpec{
+				Repositories: []v1alpha1.SpecRepository{{URL: ""}},
+			},
+		},
+	}
+
+	lg.Infof("[liveness-probe] Creating liveness probe addonsConfiguration in %q namespace", probeNamespace)
+	err := client.Create(context.TODO(), addonsConfiguration)
```

Comment:
you already have the context in `req *http.Request` just pass that into this function